src/SPC/builder/linux/LinuxBuilder.php

```diff
@@ -92,25 +87,26 @@ public function buildPHP(int $build_target = BUILD_TARGET_NONE): void
         $enableEmbed = ($build_target & BUILD_TARGET_EMBED) === BUILD_TARGET_EMBED;
         $enableFrankenphp = ($build_target & BUILD_TARGET_FRANKENPHP) === BUILD_TARGET_FRANKENPHP;
 
-        $mimallocLibs = $this->getLib('mimalloc') !== null ? BUILD_LIB_PATH . '/mimalloc.o ' : '';
         // prepare build php envs
+        $config = (new SPCConfigUtil($this, ['libs_only_deps' => true]))->config($this->ext_list, $this->lib_list, $this->getOption('with-suggested-exts'), $this->getOption('with-suggested-libs'));
+        if (str_contains($config['libs'], ' -lpgport') && !$this->getExt('pgsql')?->isBuildStatic()) {
+            // -lpgport defines many glibc functions if they are missing, which leads to feature tests succeeding that aren't meant to succeed
+            $config['libs'] = str_replace(' -lpgport', '', $config['libs']);
+        }
         $envs_build_php = SystemUtil::makeEnvVarString([
             'CFLAGS' => getenv('SPC_CMD_VAR_PHP_CONFIGURE_CFLAGS'),
             'CPPFLAGS' => getenv('SPC_CMD_VAR_PHP_CONFIGURE_CPPFLAGS'),
             'LDFLAGS' => '-L' . BUILD_LIB_PATH,
-            'LIBS' => $mimallocLibs . getenv('SPC_CMD_VAR_PHP_CONFIGURE_LIBS'),
+            'LIBS' => $config['libs'],
```

Comment:
Ah this was what introduces the various feature checks that should not succeed. We should only give those libs to the make command, not to configure.